metadata-app/src/api/index.js

```diff
@@ -88,14 +106,44 @@ const serializeSupplementalValues = (opts) => {
     }, '');
   }
 
+  const indexOfContactName = newOpts.extras.findIndex((x) => x.key === 'contact_name');
+  if (indexOfContactName > -1) {
+    newOpts.extras[indexOfContactName].value = newOpts.contact_name;
+  }
+
+  const indexOfContactEmail = newOpts.extras.findIndex((x) => x.key === 'contact_email');
+  if (indexOfContactEmail > -1) {
+    newOpts.extras[indexOfContactEmail].value = newOpts.contact_email;
+  }
+
+  const indexOfUniqueId = newOpts.extras.findIndex((x) => x.key === 'unique_id');
+  if (indexOfUniqueId > -1) {
+    newOpts.extras[indexOfUniqueId].value = newOpts.unique_id;
+  }
+
+  const indexOfPublicAccessLevel = newOpts.extras.findIndex((x) => x.key === 'public_access_level');
+  if (indexOfPublicAccessLevel > -1) {
+    newOpts.extras[indexOfPublicAccessLevel].value = newOpts.public_access_level;
+  }
+
```

Comment:
Probably we can make this in a more dynamic way, like iterating over an array of the keys such as `['public_access_level', ...] ` . Also can use find instead of `findInex()` and directly make the changes to that returned element came from `find()`. But happy either way 👍

Reply:
@kmanaseryan yeah but problem is that you can't change the object in the array. There are ways of doing it but still this looked to me obvious and clear to understand.